Potentially leaky tests

When I was contributing I noticed a strange behavior when testing. It can be easily reproduced:
```
pytest
```

this is green, but this:
```
pytest -k test_two_endpoints_with_the_same_path
```

gives
```
FAILED tests/test_spec.py::test_two_endpoints_with_the_same_path - RuntimeError: Working outside of application context.
```

I don't have much time to investigate, but it is something you might want to take a look at as it indicates that at least one test relies on a global scope.

--- a/tests/test_spec.py
+++ b/tests/test_spec.py
@@ -162,7 +162,8 @@ def test_spec_bypass_mode():
 def test_two_endpoints_with_the_same_path():
     app = create_app()
     api.register(app)
-    spec = api.spec
+    with app.app_context():
+        spec = api.spec
 
     http_methods = list(spec["paths"]["/lone"].keys())
     http_methods.sort()

